plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -202,8 +207,16 @@ rules recorder plugin = do
     liftIO $ argsSettings flags
 
   action $ do
-    files <- getFilesOfInterestUntracked
-    void $ uses GetHlintDiagnostics $ Map.keys files
+    files <- Map.keys <$> getFilesOfInterestUntracked
+    Shake.ShakeExtras{ideTesting = Options.IdeTesting testing, lspEnv} <- Shake.getShakeExtras
+    let signal :: KnownSymbol s => Proxy s -> Action ()
+        signal msg = when testing $ liftIO $ Shake.mRunLspT lspEnv $
+            LSP.sendNotification (LSP.SMethod_CustomMethod msg) $
+            toJSON $ map fromNormalizedFilePath files
+
```

Comment:
Please extract to some location. Probably to ghcide?

Reply:
would hls-test-utils be more suitable?

Reply:
I don't think so, then each plugin would need to depend on `hls-test-utils` right?

Reply:
ah, makes sense.

Reply:
seems like using `ghcide/src/Development/IDE/GHC/Util.hs` causes a cycle in the module graph, would creating a new file be better?

Reply:
Yeah, that looks like a non-ideal location. It should be in the `LSP` namespace in ghcide, perhaps `ghcide/src/Development/IDE/Core/Shake.hs`?